pageserver/src/tenant/storage_layer.rs

```diff
@@ -81,6 +86,54 @@ pub enum ValueReconstructResult {
     Missing,
 }
 
+#[derive(Default)]
+pub struct LayerAccessStats(RwLock<LayerAccessStatsInner>);
+
+#[derive(Default)]
+struct LayerAccessStatsInner {
+    last_access: Option<SystemTime>,
+    first_access: Option<SystemTime>,
```

Comment:
With this mutex construction we can just use Instants everywhere.

Reply:
`Instant` has no meaning outside of the process in which it was generated (`CLOCK_MONOTONIC`).
We want to communicate this timestamp to the outside world, so, a wall clock timestamp is needed.

That being said, we could store `Instant`s internally and change the public  API to return

```
age: Duration, // Instant::now() - first_access 
not_accessed_for: Duration // Instant::how() - last_access
```

I think for a dump endpoint, absolute timestamps are better, though.

For a human playing with this, relative times as proposed above would actually be more intuitive.

Reply:
Aah yes, the conversions I was expecting to find were not there. Resolving.